launchable/manager.py

```diff
@@ -68,6 +76,37 @@ def _reorder(test, order):
             tests.append(node)
             tree_nodes.append((node, child))
 
-    test._tests = tests
+    suite._tests = tests
 
     return tree_nodes
+
+
+# Check if test's context is a leaf
+def _is_leaf(suite):
```

Comment:
This is the single most important function I added in this PR. There are three possible cases we need to stop loading as I described below comment:

1. When the context is Failure, that means nose fails to load the specified test file the node will be a leaf and we don't need to go down further.
2. If the context is Module, there are two possibilities 1. the module is a package 2. the module is a file. If the module is a package i.e. `__file__` ends with `__init__.py` we need to go down further and look for a file.
3. If the context is a class we should stop loading there. This is a relatively rare case when a user specifies a class name directly

Reply:
This function is very clear. I can read your purpose easily from this code. I love this function ❤️